keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1116 @@
+for both the Admin and Developer will need to be created. For the Admin, it is 
+safe to assume they will have the correct RBAC roles to list all the NetworkPolicies 
+and AdminNetworkPolicies in a cluster. Therefore, the Admin oriented tooling should 
+be able to both alert the Admin to any overriding of NetworkPolicies that may occur if a 
+new AdminNetworkPolicy is to be created and provide a warning if there is another ANP 
+with the same priority. For the Developer, who usually can only list the NetworkPolicies 
+in a given namespace, the tooling should simply alert if a given NetworkPolicy would
+be overridden by any of the ANPs in a cluster. The aforementioned tooling will not 
+be a primary development goal during the alpha version of this API, and will most 
+likely be completed during the beta development cycle. 
+
+### Future Work
+
+Although the scope of the AdminNetworkPolicies is extensive, the above proposal
+intends to only solve the documented use cases. However, we would
+also like to consider the following set of proposals as future work items:
+- **Audit Logging**: Very often cluster administrators want to log every connection
+  that is either denied or allowed by a firewall rule and send the details to
+  an IDS or any custom tool for further processing of that information.
+  With the introduction of `deny` rules, it may make sense to incorporate the
+  cluster-scoped policy resources with a new field, say `auditPolicy`, to
+  determine whether a connection matching a particular rule/policy must be
+  logged or not.
+
+## Design Details
+
+### AdminNetworkPolicy API Design
+
+The following new `AdminNetworkPolicy` API will be added to the `policy.networking.k8s.io` 
+API group.  
+
+```golang
+type AdminNetworkPolicy struct {
+  metav1.TypeMeta
+  metav1.ObjectMeta
+  // Specification of the desired behavior of AdminNetworkPolicy.
+  Spec     AdminNetworkPolicySpec
+  // Status to be reported by the CNI, this is not standardized in alpha and 
+  // CNIs should report what they see fit in relation to their AdminNetworkPolicy
+  // implementation
+  // +optional 
+  Status   []metav1.Condition
+}
+
+type AdminNetworkPolicySpec struct {
+  // Int value bound to 0 - 999, with 0 signaling the special "baseline" priority
+  Priority      Int 
+  // Allows for multiple AdminNetworkPolicy rules at a single priority level
+  // If empty no rules are applied
+  // A total of 100 total rules will be allowed per each network policy instance, 
+  // this rule count will be calculated as the total summation of the 
+  // AdminNetworkPolicyRules, AdminNetworkPolicyIngressRules, and AdminNetworkPolicyEgressRules
+  // in a single AdminNetworkPolicy Instance. 
+  PolicyRules   []AdminNetworkPolicyRule
+}
+
+type AdminNetworkPolicyRule struct {
+  // No implicit isolation of AppliedTo Pods/Namespaces. 
+  // If no rules are a defined no behavior will be observed 
+  AppliedTo   AppliedTo
```

Comment:
I started a thread with  api-reviewers on a convention for this.  I proposed "target" but "subject" may be a better choice.

Reply:
Do you have a link to that thread?

Reply:
Changed this to subject

Reply:
https://groups.google.com/g/kubernetes-api-reviewers/c/ldmrXXQC4G4